lib/client.js

```diff
@@ -221,13 +221,15 @@ function processResponse(path, result) {
 
 function isOriginValid(origin) {
   var WHITELISTED_ORIGINS = [
-    /.*zendesk\.com.*/gi,
-    /.*zd-staging\.com.*/gi,
-    /.*zd-dev\.com.*/gi,
-    /.*zd-master\.com.*/gi,
-    /.*zendesk-staging\.com.*/gi,
-    /.*dashboard\.zopim\.com.*/gi,
-    /.*dashboard\.zopim\.org.*/gi
+    /^http(s)?:\/\/127.0.0.1(:\d+)?$/,
+    /^http(s)?:\/\/localhost(:\d+)?$/,
+    /^https:\/\/\w+\.zendesk\.com$/,
+    /^https:\/\/\w+\.zd-staging\.com$/,
+    /^https:\/\/\w+\.zd-dev\.com$/,
+    /^https:\/\/\w+\.zd-master\.com$/,
+    /^https:\/\/\w+\.zendesk-staging\.com$/,
+    /^https:\/\/dashboard\.zopim\.com$/,
```

Comment:
Nice, these checks are much more specific now 👍 Btw, I just realised that this won't work for local dev env for Chat though because they might use something like `<custom_local>.zopim.com`, i.e. I'm using `https://hoanglocal.zopim.com:8080`